fluent/protocol/transport.go

```diff
@@ -139,6 +142,215 @@ type EntryExt struct {
+	if arrayLen != 2 {
+		return msgp.ArrayError{Wanted: 2, Got: arrayLen}
+	}
+
+	// peek at the next msgpack type to determine timestamp format
+	nextType, err := r.NextType()
+	if err != nil {
+		return msgp.WrapError(err, "Timestamp peek")
+	}
+
+	switch nextType {
+	case msgp.ExtensionType:
+		// extension format
+		err = r.ReadExtension(&z.Timestamp)
+		if err != nil {
+			return msgp.WrapError(err, "Timestamp extension")
+		}
+	case msgp.ArrayType:
+		// timestamp with metadata format: [timestamp, metadata]
+		var arrSize uint32
+		arrSize, err = r.ReadArrayHeader()
+		if err != nil {
+			return msgp.WrapError(err, "Timestamp array header")
+		}
+
+		if arrSize < 1 || arrSize > 2 {
+			return msgp.WrapError(err, "Invalid timestamp array size")
+		}
+
+		nt, err := r.NextType()
+		if err != nil {
+			return msgp.WrapError(err, "Timestamp peek")
+		}
+
+		switch nt {
+		case msgp.ExtensionType:
+			err = r.ReadExtension(&z.Timestamp)
+			if err != nil {
+				return msgp.WrapError(err, "Timestamp extension from array")
+			}
+		case msgp.IntType:
+			// read the timestamp value
+			var seconds int64
+			seconds, err = r.ReadInt64()
+			if err != nil {
+				return msgp.WrapError(err, "Timestamp from array")
+			}
+
+			z.Timestamp = EventTime{time.Unix(seconds, 0).UTC()}
+		}
+
+		// Skip metadata elements (all elements after the first)
+		// We iterate starting from 1 because element 0 (timestamp) is already read
+		for i := uint32(1); i < arrSize; i++ {
+			// Skip() advances past one complete msgpack object in the byte slice
+			err = r.Skip()
+			if err != nil {
+				return msgp.WrapError(err, "Skip metadata element")
+			}
+		}
+
+	case msgp.IntType:
+		// this is simple timestamp format: unix seconds as integer
+		var sec int64
+		sec, err = r.ReadInt64()
+		if err != nil {
+			return msgp.WrapError(err, "Timestamp int")
+		}
+		z.Timestamp = EventTime{time.Unix(sec, 0).UTC()}
+
+	case msgp.UintType:
+		// unsigned integer timestamp
+		var usec uint64
+		usec, err = r.ReadUint64()
+		if err != nil {
+			return msgp.WrapError(err, "Timestamp uint")
+		}
+		if usec > math.MaxInt64 {
+			return msgp.WrapError(err, "timestamp overflow exceeds maximum int64 value")
+		}
+		z.Timestamp = EventTime{time.Unix(int64(usec), 0).UTC()} /* #nosec G115 */
+
+	default:
+		return msgp.WrapError(err, "unsupported timestamp type: expected int or array")
+	}
+
+	// rad the record (second element)
+	z.Record, err = r.ReadIntf()
+	if err != nil {
+		return msgp.WrapError(err, "Record")
+	}
+
+	return nil
+}
+
+// This decodes msgpack data from a byte slice into EntryExt
+// - [timestamp, record]
+// - where timestamp can be an integer or [integer, {metadata}] (we ignore metadata)
+func (z *EntryExt) UnmarshalMsg(bts []byte) ([]byte, error) {
+	// read array header to get number of elements
+	arrayLen, bts, err := msgp.ReadArrayHeaderBytes(bts)
+	if err != nil {
+		err = msgp.WrapError(err, "Array Header")
```

Comment:
The linter seems to force you in fmt.Errorf("") to start with lowercase letters. Why not also with `msgp.WrapError()` - just because the linter doesn't know about it?

Reply:
And do you think the existing WrapError message style is correct? I vote for a common style - regardless if the linter only knows one of the two interfaces.